numpy/core/src/multiarray/alloc.c

```diff
@@ -326,31 +326,40 @@ PyDataMem_RENEW(void *ptr, size_t size)
 }
 
 /* Memory handler global default */
-PyDataMem_Handler default_allocator = {
+PyDataMem_Handler default_handler = {
     "default_allocator",
-    npy_alloc_cache,      /* alloc */
-    npy_alloc_cache_zero, /* zeroed_alloc */
-    npy_free_cache,       /* free */
-    PyDataMem_RENEW       /* realloc */
+    {
+        NULL, /* ctx */
+        NULL, /* (npy_alloc_cache) malloc */
+        NULL, /* (npy_alloc_cache_zero) calloc */
+        NULL, /* (PyDataMem_RENEW) realloc */
+        NULL  /* (npy_free_cache) free */
+    }
```

Comment:
The default allocator routines (which are also used for internal allocations) do not accept a `ctx`.

Since `PyDataMem_UserNEW/NEW_ZEROED/RENEW/FREE` already had special handling for them, I simplified the way the default allocator is handled (#L357, #L380, #L402, #L423).